platform/mbed_sleep.h

```diff
@@ -123,7 +127,9 @@ __INLINE static void sleep(void)
 {
 #if !(defined(FEATURE_UVISOR) && defined(TARGET_UVISOR_SUPPORTED))
 #if DEVICE_SLEEP
+#if (MBED_CONF_RTOS_PRESENT == 0) || (DEVICE_STCLK_OFF_DURING_SLEEP == 0) || defined(MBED_TICKLESS)
```

Comment:
- Should this be more descriptive `DEVICE_SYSTICKCLK.....` or `DEVICE_SYSTCLK....` ? If I did not read the above comment, I would be guessing what ST clk stands for.

Reply:
STCLK is used in the ARM architecture manuals. I choose to stick with it in the sake of consistency but I can change it.